test/RevokableTokenLock.spec.ts

```diff
@@ -0,0 +1,125 @@
+      await expect(revokableTokenLock.setRevoker(other.address)).to.not.be.reverted;
+    });
+    it('should update value of revoker in contract to other address', async () => {
+      await revokableTokenLock.setRevoker(other.address);
+      const revoker = await revokableTokenLock.revoker();
+      expect(revoker).to.equal(other.address);
+    });
+  });
+  describe('#revoke', async () => {
+    let transferred: BigNumber;
+    let remaining: BigNumber;
+    let unlockBegin: BigNumber;
+    let unlockCliff: BigNumber;
+    let unlockEnd: BigNumber;
+    let claimedAmounts: BigNumber;
+    let lockedAmounts: BigNumber;
+    before('Set up data', async () => {
+      ({unlockBegin, unlockCliff, unlockEnd} = await revokableTokenLock.vesting(recipient.address));
+      transferred = amount.mul(unlockCliff.sub(unlockBegin)).div(unlockEnd.sub(unlockBegin));
+      remaining = amount.sub(transferred);
+    });
+    it('should revert if caller is not owner or revoker', async () => {
+      await expect(revokableTokenLock.connect(other).revoke(recipient.address)).to.be.revertedWith(
+        'RevokableTokenLock: onlyAuthorizedActors'
+      );
+    });
+    it('should not revert if caller is revoker', async () => {
+      await expect(revokableTokenLock.connect(revoker).revoke(recipient.address)).to.not.be.reverted;
+    });
+    it('should not revert if caller is owner', async () => {
+      await expect(revokableTokenLock.revoke(recipient.address)).to.not.be.reverted;
+    });
+    it('should claim none and revoke all tokens to owner if revoke is called before unlockCliff', async () => {
+      await expect(() => revokableTokenLock.revoke(recipient.address)).to.changeTokenBalances(
+        token,
+        [recipient, owner],
+        [0, amount]
+      );
+    });
+    it('should claim any tokens vested and revoke the remaining to owner when revoke is called after unlockCliff but before unlockEnd', async () => {
+      await setNextBlockTimeStamp(unlockCliff.toNumber());
+      await expect(() => revokableTokenLock.revoke(recipient.address)).to.changeTokenBalances(
+        token,
+        [recipient, owner],
+        [transferred, remaining]
+      );
+    });
+    it('should claim all tokens and revoke none to owner when revoke is called after unlockEnd', async () => {
+      await setNextBlockTimeStamp(unlockEnd.toNumber());
+      await expect(() => revokableTokenLock.revoke(recipient.address)).to.changeTokenBalances(
+        token,
+        [recipient, owner],
+        [amount, 0]
+      );
+    });
+    it('should return claimable balance of 0 after some time has passed since revoke was called.', async () => {
+      await setNextBlockTimeStamp(unlockCliff.toNumber());
+      await revokableTokenLock.revoke(recipient.address);
+
+      // Set next block timestamp to 2 hour after revoke was called and mine a block.
+      await setNextBlockTimeStamp(unlockCliff.add(HOUR.mul(TWO)).toNumber());
+      await mineBlocks(ONE.toNumber());
```

Comment:
There's a shared `mineAtBlock` function in the `TokenLock` test PR. If this is merged, might just want to use this one instead. https://github.com/code-423n4/contracts/pull/29/files#diff-6d7a55d4a3b08bd886254229dc52ae7929824c0c293304d75ae168864c43b0edR7

Reply:
updated!